### ISSUE TITLE
Riposte without Riposte

### ISSUE THREAD

**Issue by ennvina** [ennvina/spellactivationoverlay#56]
Even without the Riposte talent, the Riposte SAO shows up after a parry.

It probably should not happen because we track the spell based on `IsUsableSpell` which, in theory, should return false because the player doesn’t know the spell.

**Comment by ennvina** [comment]
Suggestion to solve this issue:
- add a `talent` parameter to the RegisterCounter method, e.g.
```lua
function SAO.RegisterCounter(self, auraID, talent)
```
- if `talent` is nil, talent is not checked (= current behavior)
- if `talent` is not nil, it should be an object `{ tab, index }` and will be checked before every proc
- `tab` and `index` can be fetched with `SAO:GetTalentByName(spellName)`

This would be an addition to the Counter system in `counter.lua`.

Then, in `rogue.lua`, call something like this:
```lua
    local riposteSpellID = 14251;
    local riposteSpellName = GetSpellInfo(riposteSpellID);
    local _, _, tab, index = SAO:GetTalentByName(riposteSpellName);
    self:RegisterAura("riposte", 0, riposteSpellID, "bandits_guile", "Top (CW)", 1, 255, 255, 255, true, { riposteSpellID });
    self:RegisterCounter("riposte", { tab, index }); -- Must match name from above call
```

**Comment by ennvina** [comment]
Implemented in #73.
